integration/hive/src/main/java/org/apache/carbondata/hive/util/HiveCarbonUtil.java

```diff
@@ -171,15 +171,33 @@ public static CarbonTable getCarbonTable(Configuration tableProperties) throws S
       columns = columns + "," + partitionColumns;
       columnTypes = columnTypes + ":" + partitionColumnTypes;
     }
-    String[] columnTypeArray = HiveCarbonUtil.splitSchemaStringToArray(columnTypes);
-
+    String[][] validatedColumnsAndTypes = validateColumnsAndTypes(columns, columnTypes);
     CarbonTable carbonTable = CarbonTable.buildFromTableInfo(
         HiveCarbonUtil.getTableInfo(tableName, databaseName, tablePath,
-            sortColumns, columns.split(","), columnTypeArray, new ArrayList<>()));
+            sortColumns, validatedColumnsAndTypes[0],
+            validatedColumnsAndTypes[1], new ArrayList<>()));
     carbonTable.setTransactionalTable(false);
     return carbonTable;
   }
 
+  private static String[][] validateColumnsAndTypes(String columns, String columnTypes) {
+    String[] columnTypeArray = HiveCarbonUtil.splitSchemaStringToArray(columnTypes);
```

Comment:
what are these changes for didn't see anything about this in descriptions. please add comments

Reply:
In case of empty table some additional columns are getting added in the configuration. Here I have validated if any additional column and removed that.